src/responder/common/responder_common.c

```diff
@@ -126,9 +123,9 @@ static errno_t get_client_cred(struct cli_ctx *cctx)
         return ENOMSG;
     }
 
-    if (cctx->creds->ucred.pid > -1) {
+    if (cli_creds_get_pid(cctx->creds) > -1) {
```

Comment:
I wonder if because of this check we should actually leave the `cctx->creds->ucred.pid = -1;` in place?

Reply:
This is a kind of a sanity check, that imo doesn't make much sense.

If `getsockopt()` fails, code doesn't get here.
If  `getsockopt()` doesn't fail, it should fill `ucred`, overwriting '-1' initialization anyway.